decode/evm_rpc.go

```diff
@@ -6,12 +6,15 @@ import (
 	"errors"
 	"fmt"
 
-	"github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/ethclient"
-
 	cosmosmath "cosmossdk.io/math"
+	"github.com/ethereum/go-ethereum/common"
+	ethctypes "github.com/ethereum/go-ethereum/core/types"
 )
 
+type EVMClient interface {
```

Comment:
comment as to what this struct is used for, and  does it really belong in this package or in pack clients?

Reply:
it's mostly used to change signature of this function:
```go
func (r *EVMRPCRequestEnvelope) ExtractBlockNumberFromEVMRPCRequest(ctx context.Context, evmClient EVMClient) (int64, error)
```

to make it dependent from small interface, not from specific ethclient, it simplify things in tests

but I think it can be renamed to` BlockGetter` or smth like this, maybe `BlockByHashGetter`

Reply:
@galxy25 reimplemented in such way:
```go
// EVMBlockGetter defines an interface which can be implemented by any client capable of getting ethereum block by hash
type EVMBlockGetter interface {
	// BlockByHash returns ethereum block by hash
	BlockByHash(ctx context.Context, hash common.Hash) (*ethctypes.Block, error)
}
```